.github/workflows/qa-tests.yaml

```diff
@@ -0,0 +1,45 @@
+name: QA Tests
+
+on:
+  pull_request:
+
+jobs:
+  CI:
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install the Apple certificate and provisioning profile
+        env:
+          BUILD_CERTIFICATE_BASE64: ${{ secrets.IOS_BUILD_CERTIFICATE_BASE64 }}
+          P12_PASSWORD: ${{ secrets.P12_PASSWORD }}
+          BUILD_PROVISION_PROFILE_BASE64: ${{ secrets.IOS_BUILD_PROVISION_PROFILE_BASE64 }}
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+        run: |
+          # create variables
+          CERTIFICATE_PATH=$RUNNER_TEMP/build_certificate.p12
+          PP_PATH=$RUNNER_TEMP/build_pp.mobileprovision
+          KEYCHAIN_PATH=$RUNNER_TEMP/app-signing.keychain-db
+          # import certificate and provisioning profile from secrets
+          echo -n "$BUILD_CERTIFICATE_BASE64" | base64 --decode -o $CERTIFICATE_PATH
+          echo -n "$BUILD_PROVISION_PROFILE_BASE64" | base64 --decode -o $PP_PATH
+          # create temporary keychain
+          security create-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security set-keychain-settings -lut 21600 $KEYCHAIN_PATH
+          security unlock-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          # import certificate to keychain
+          security import $CERTIFICATE_PATH -P "$P12_PASSWORD" -A -t cert -f pkcs12 -k $KEYCHAIN_PATH
+          security list-keychain -d user -s $KEYCHAIN_PATH
+          # apply provisioning profile
+          mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+          cp $PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+      - name: Build app
+        run: |
+          ls ~/Library/MobileDevice/Provisioning\ Profiles
+          cd Sample
+          mkdir -p ios_build
+          swift build
+          xcodebuild -scheme SampleForageSDK -destination 'generic/platform=iOS' CONFIGURATION_BUILD_DIR=./ios_build build DEVELOPMENT_TEAM=3D54TBBRF3 -allowProvisioningUpdates CODE_SIGNING_ALLOWED=NO
+          ls
```

Comment:
Clarifying question: does this job run a QA-Tests suite? I followed that it builds the SampleForageSDK but wasn't able to discern whether a test suite gets run. Thanks for any clarity!

Reply:
It will be doing that once it's done and working :) battling with iOS and github actions is not an easy task.

Reply:
I'm moving the QA test work to a separate PR so I will merge this to start the linting step.